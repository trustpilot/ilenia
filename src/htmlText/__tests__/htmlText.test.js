import React from 'react';
import renderer from 'react-test-renderer';
import { LocalizationProvider, HtmlText } from '../../';

const translations = {
  'test1': 'Just a random string',
  'test2': 'A string but with a [HTML] in it',
  'test3': 'A string that is sort of [HTML]italic[HTML] and [HTML]bold[HTML] and such',
  'test4': 'A string that is sort of [HTML]italic[HTML]',
  'test5': 'A string that uses {smtg}another{smtg} key'
};

const setupText = (string, interpolations, key) => (<LocalizationProvider locale="en-US" translations={translations}>
  <HtmlText id={string} interpolations={interpolations} key={key} />
</LocalizationProvider>);

test('Renders a plain string', () => {
  const component = renderer.create(setupText('test1'));

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Renders a string with one HTML part', () => {
  const component = renderer.create(setupText('test2', ['<span>hello</span>']));

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Renders a string with multiple HTML parts in it', () => {
  const component = renderer.create(setupText('test3', ['<em>', '</em>', '<b>', '</b>']));

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Renders a string with HTML parts in the end', () => {
  const component = renderer.create(setupText('test4', ['<em>', '</em>']));

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Renders a string with HTML marked with another key', () => {
  const component = renderer.create(setupText('test5', ['<b>', '</b>'], '{smtg}'));

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
