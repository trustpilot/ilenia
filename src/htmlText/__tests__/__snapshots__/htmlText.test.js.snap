// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders a plain string 1`] = `
<span
  dangerouslySetInnerHTML={
    Object {
      "__html": "Just a random string",
    }
  }
/>
`;

exports[`Renders a string with HTML marked with another key 1`] = `
<span
  dangerouslySetInnerHTML={
    Object {
      "__html": "A string that uses {smtg}another{smtg} key<b>",
    }
  }
/>
`;

exports[`Renders a string with HTML parts in the end 1`] = `
<span
  dangerouslySetInnerHTML={
    Object {
      "__html": "A string that is sort of <em>italic</em>",
    }
  }
/>
`;

exports[`Renders a string with multiple HTML parts in it 1`] = `
<span
  dangerouslySetInnerHTML={
    Object {
      "__html": "A string that is sort of <em>italic</em> and <b>bold</b> and such",
    }
  }
/>
`;

exports[`Renders a string with one HTML part 1`] = `
<span
  dangerouslySetInnerHTML={
    Object {
      "__html": "A string but with a <span>hello</span> in it",
    }
  }
/>
`;
